src/main/java/model/process/analysis/operations/comparisons/TimeBetween.java

```diff
@@ -1,11 +1,168 @@
 package model.process.analysis.operations.comparisons;
 
+import java.time.Duration;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.Period;
+import java.util.InputMismatchException;
+
+import model.data.DataColumn;
+import model.data.DataRow;
+import model.data.DataTable;
+import model.data.DataTableConversionBuilder;
+import model.data.Table;
+import model.data.value.DataValue;
+import model.data.value.DateTimeValue;
+import model.data.value.DateValue;
+import model.data.value.PeriodValue;
+import model.data.value.TimeValue;
+import model.language.Identifier;
+import model.process.DataProcess;
+
 /**
- * This class will determine the time between events.
- * After they have been put in chronological order by LSA.
+ * This class will determine the time difference between rows. This is actually
+ * a DataProcess but it's listed under comparisons in the requirements. ^ The
+ * reason why it's located in the analysis package.
  * 
- * @author Louis Gosschalk 27-05-2015
+ * @author Louis Gosschalk 12-06-2015
  */
-public class TimeBetween {
+public class TimeBetween extends DataProcess {
+
+	private String dateName;
+	private DataValue value;
+	private LocalDate curdate;
+	private LocalTime curtime;
+	private LocalDate predate;
+	private LocalTime pretime;
+	private int hour;
+	private int min;
+	private int sec;
+
+	public TimeBetween(Identifier<DataColumn> date) {
+		this.dateName = date.getName();
+	}
+
+	/**
+	 * Check the type of the given column and call calculation.
+	 */
+	@Override
+	public Table doProcess() {
+		value = ((DataTable) getInput()).getRow(0).getValue(
+				((DataTable) getInput()).getColumn(dateName));
+
+		DataTableConversionBuilder builder = new DataTableConversionBuilder(
+				((DataTable) getInput()), ((DataTable) getInput()).getName());
+		builder.addRowsFromTable((DataTable) getInput());
+
+		if (!value.getClass().equals(TimeValue.class)) {
+			builder.addColumn((DataTable) getInput(), new PeriodValue(null,
+					null, null), "Difference date");
+		}
+		if (!value.getClass().equals(DateValue.class)) {
+			builder.addColumn((DataTable) getInput(), new TimeValue(null, null,
+					null), "Difference time");
+		}
+
+		setInput(builder.build());
+
+		DataTable result = getVars((DataTable) getInput());
+
+		return result;
+
+	}
+
+	private DataTable getVars(DataTable table) {
```

Comment:
Ik heb dit even geprobeerd maar het kan helaas niet in minder lines (volgens mij) vanwege het casten. Als ik dingen ga samenvoegen (zoals de dateTime opbouwen in date en time zodat er maar 2 if statements zij ipv 3) dan krijg je cast-errors